src/unity/toolkits/activity_classification/sequence_iterator.cpp

```diff
@@ -24,27 +24,29 @@ static std::map<std::string,size_t> generate_column_index_map(const std::vector<
 
 /**
  *  Find the statistical mode (majority value) of a given vector.
- *  Based on https://en.wikipedia.org/wiki/Boyer–Moore_majority_vote_algorithm
  *
  * \param[in] input_vec a vector for which the mode will be calculated
 
  * \return    The most frequent value within the given vector.
  */
-static double vec_majority_value(const flex_vec& input_vec) {
-    size_t counter = 0;
-    double candidate = 0.0;
+
+static double vec_mode(const flex_vec& input_vec) {
+    std::vector<int> histogram;
     for (size_t i = 0; i < input_vec.size(); ++i) {
-        double value = input_vec[i];
-        if (counter == 0) {
-            candidate = value;
-            counter = 1;
-        } else if (candidate == value) {
-            counter++;
-        } else {
-            counter--;
+        int value = static_cast<int>(input_vec[i]);
+
+        assert(static_cast<double>(static_cast<double>(input_vec[i])) == input_vec[i]);
```

Comment:
I think you want the inner cast here to return `int`, since casting to `double` twice is redundant. Might also be worth a simple comment, something like "Each value should be in the index of a class label."